### DISCUSSION TITLE
Iframes

### DISCUSSION THREAD

**Discussion by sHermanGriffiths** [boisgera/pandoc#61] [Q&A] [answered]
Is there any class that represents an iframe? I'd like to output an iframe in html and markdown.

**Comment by boisgera** [comment, marked as answer]
There is no Iframe class in the Python pandoc library since there is no such type in the [Pandoc Haskell collection of types](https://hackage.haskell.org/package/pandoc-types-1.23.1/docs/Text-Pandoc-Definition.html).

But if you want to include iframes in a pandoc document (programatically) you can use the `RawBlock` class. An example would be:

```python
import pandoc
from pandoc.types import Pandoc, Meta, RawBlock, Header, Str

header = Header(1, ["", [], []], [Str("OSM")])

iframe = RawBlock("html",
"""
<iframe
  id="inlineFrameExample"
  title="Inline Frame Example"
  width="300"
  height="200"
  src="https://www.openstreetmap.org/export/embed.html?bbox=-0.004017949104309083%2C51.47612752641776%2C0.00030577182769775396%2C51.478569861898606&layer=mapnik"
>
</iframe>
"""
)


doc = Pandoc(Meta({}), [header, iframe])

print(pandoc.write(doc, format="html"))

print(pandoc.write(doc, format="markdown"))
```

The same trick works for LaTeX constructs that are not directly supported as types.
